src/main/java/org/example/bookshop/repository/SpecificationProviderManagerImpl.java

```diff
@@ -0,0 +1,22 @@
+package org.example.bookshop.repository;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class SpecificationProviderManagerImpl<T>
+        implements SpecificationProviderManager<T> {
+
+    private final List<SpecificationProvider<T>> bookSpecificationProviders;
+
+    @Override
+    public SpecificationProvider<T> getSpecificationProvider(String key) {
+        return bookSpecificationProviders.stream()
+            .filter(b -> b.getKey().equals(key))
+            .findFirst()
+            .orElseThrow(() -> new RuntimeException(
+                    "There is no SpecificationProvider with this key" + key));
```

Comment:
use provider instead of short naming b
use NoSuchElementException or create a custom exception